fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java

```diff
@@ -0,0 +1,308 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.load;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Range;
+import com.starrocks.analysis.InsertStmt;
+import com.starrocks.analysis.PartitionNames;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Partition;
+import com.starrocks.catalog.PartitionInfo;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.catalog.PartitionType;
+import com.starrocks.catalog.RangePartitionInfo;
+import com.starrocks.persist.AddPartitionsInfo;
+import com.starrocks.persist.InsertOverwriteStateChangeInfo;
+import com.starrocks.persist.PartitionPersistInfo;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.QueryState;
+import com.starrocks.qe.StmtExecutor;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.StatementPlanner;
+import com.starrocks.sql.plan.ExecPlan;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InsertOverwriteJobRunner {
+    private static final Logger LOG = LogManager.getLogger(InsertOverwriteJobRunner.class);
+
+    private InsertOverwriteJob job;
+
+    private InsertStmt insertStmt;
+    private StmtExecutor stmtExecutor;
+    private ConnectContext context;
+    private Database db;
+    private OlapTable targetTable;
+    private String postfix;
+
+    private long createPartitionElapse;
+    private long insertElapse;
+
+    public InsertOverwriteJobRunner(InsertOverwriteJob job, ConnectContext context, StmtExecutor stmtExecutor,
+                                    Database db, OlapTable targetTable) {
+        this.job = job;
+        this.context = context;
+        this.stmtExecutor = stmtExecutor;
+        this.insertStmt = job.getInsertStmt();
```

Comment:
because you will change this insertStmt, you should deep copy it.

Reply:
I just change the TargetPartitionNames and TargetPartitionIds, which are just some names and ids, generated by parser, and not something stored in meta. And I test that these fields do not impact the result of show process list. And more it seems not easy to deep copy InsertStmt now.
I think it is ok to modify some fields of InsertStmt.